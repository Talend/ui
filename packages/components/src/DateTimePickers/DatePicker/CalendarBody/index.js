import CalendarBody from './CalendarBody.component';

export default CalendarBody;
