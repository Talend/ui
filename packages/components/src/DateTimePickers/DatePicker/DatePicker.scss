@import '../common';

.container {
    @include main;
    display: flex;
    height: 100%;
    flex-direction: column;
    justify-content: space-evenly;

    .calendar-row {
        display: flex;
    }

    .calendar-item {
        display: flex;
        flex: 1;
        justify-content: center;
        align-content: center;
    }

    .calendar-header-row {
        color: $gray;

        abbr[title] {
           font-variant: none;
           border: none;
           text-decoration: none;
        }
    }

    .separator {
        width: calc(100% - 20px);
        border: none;
        border-bottom: $border-default;
        margin: 0 auto;
    }

    .calendar-body-row {
        .calendar-item {
            .calendar-day-action {
                background: none;
                border: none;
                width: 2em;
                height: 2em;
                border-radius: 50%;

                &:hover {
                    background-color: $concrete;
                }

                &:disabled {
                    color: $silver-chalice;
                    background-color: transparent;
                }
            }

            &.today .calendar-day-action  {
                background-color: rgba($slate-gray, 0.12);
            }

            &.active .calendar-day-action {
                @include picker-action-active;
            }
        }
    }
}
