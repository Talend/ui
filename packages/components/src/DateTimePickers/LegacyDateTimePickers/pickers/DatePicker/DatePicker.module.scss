@use '~@talend/bootstrap-theme/src/theme/guidelines' as *;
@use '~@talend/design-tokens/lib/tokens';

@import '../../shared/styles/mixins';
@import '../../shared/styles/variables';

.container {
	width: 100%;
	position: relative;
}

.calendar-header {
	&::after {
		@include picker-underline;
		width: 100%;
		position: absolute;
		left: 0;
		top: 2.3rem;
	}

	th {
		text-align: center;
		padding-bottom: $padding-small;
	}

	abbr {
		border: none;
		font-weight: normal;
		text-decoration: none;
	}
}

.calendar-row {
	height: 3.6rem;
	text-align: center;

	.calendar-day {
		@include picker-action;

		height: 2.4rem;
		width: 2.4rem;
		background: transparent;
		border: none;
		border-radius: 50%;
		line-height: 2.4rem;
		padding: 0;

		&.today {
			background-color: tokens.$coral-color-accent-background;
			color: tokens.$coral-color-accent-text-strong;
			border: solid 1px tokens.$coral-color-accent-border;
		}

		&.not-current-month {
			opacity: 0.54;
		}
	}
}
