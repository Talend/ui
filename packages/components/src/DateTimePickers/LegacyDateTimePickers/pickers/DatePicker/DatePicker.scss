@import '../../shared/styles/mixins';
@import '../../shared/styles/variables';

.container {
	width: 100%;
	position: relative;
}

.calendar-header {
	&:after {
		@include picker-underline;
		width: 100%;
		position: absolute;
		left: 0;
		top: 2.3rem;
	}

	th {
		text-align: center;
		padding-bottom: $padding-small;
	}

	abbr {
		border: none;
		font-weight: normal;
		text-decoration: none;
	}
}

.calendar-row {
	height: 3.6rem;
	text-align: center;

	.calendar-day {
		&.today {
			background-color: tint($lochmara, 90%);
			color: $lochmara;
			border: solid 1px $lochmara;
		}

		@include picker-action;

		height: 2.4rem;
		width: 2.4rem;
		background: transparent;
		border: none;
		border-radius: 50%;
		line-height: 2.4rem;
		padding: 0;

		&.not-current-month {
			opacity: 0.54;
		}
	}
}
