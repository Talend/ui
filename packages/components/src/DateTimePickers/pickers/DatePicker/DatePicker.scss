@import '../../shared/styles/mixins';
@import '../../shared/styles/variables';

.container {
	&:after {
		@include picker-underline;
		bottom: -$padding-smaller;
	}
	width: 100%;
	position: relative;
}

.calendar-header {
	height: 3rem;
	&:after {
		@include picker-underline;
		top: 2.5rem;
	}

	th {
		text-align: center;
		padding-bottom: $padding-small;
	}

	abbr {
		border: none;
		font-weight: normal;
		text-decoration: none;
	}
}

.calendar-row {
	height: 3.6rem;
	text-align: center;

	.calendar-day {
		&.today {
			background-color: $concrete;
		}

		@include picker-action;

		height: 2.4rem;
		width: 2.4rem;
		background: transparent;
		border: none;
		border-radius: 50%;
		line-height: 2.4rem;
		padding: 0;
	}
}
