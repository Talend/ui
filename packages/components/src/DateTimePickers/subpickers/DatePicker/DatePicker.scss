@import '../../shared/common';

.container {
    @include main;
    display: flex;
    height: 100%;
    flex-direction: column;
    justify-content: space-evenly;

    .calendar-row {
        display: flex;
    }

    .calendar-item {
        display: flex;
        flex: 1;
        justify-content: center;
        align-content: center;
    }

    .calendar-header-row {
        color: $gray;

        abbr[title] {
           font-variant: none;
           border: none;
           text-decoration: none;
        }
    }

    .separator {
        width: calc(100% - 20px);
        border: none;
        border-bottom: $border-default;
        margin: 0 auto;
    }
}
