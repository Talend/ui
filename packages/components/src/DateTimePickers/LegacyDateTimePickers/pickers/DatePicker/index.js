import { withTranslation } from 'react-i18next';
import DatePicker from './DatePicker.component';
import I18N_DOMAIN_COMPONENTS from '../../../../constants';

export default withTranslation(I18N_DOMAIN_COMPONENTS)(DatePicker);
