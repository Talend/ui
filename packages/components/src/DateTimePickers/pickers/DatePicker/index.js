import { translate } from 'react-i18next';
import DatePicker from './DatePicker.component';
import I18N_DOMAIN_COMPONENTS from '../../../constants';

export default translate(I18N_DOMAIN_COMPONENTS)(DatePicker);
