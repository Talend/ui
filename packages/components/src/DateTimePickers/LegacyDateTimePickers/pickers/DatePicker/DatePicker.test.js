import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { isSameDay } from 'date-fns/isSameDay';
import { isToday } from 'date-fns/isToday';

import DatePicker from './DatePicker.component';

jest.mock('date-fns/isToday');

function mockIsTodayWith(newToday) {
	isToday.mockImplementation(date => isSameDay(date, newToday));
}

function getDisabledChecker(disabledDates) {
	return date => disabledDates.some(disabledDate => isSameDay(disabledDate, date));
}

describe('DatePicker', () => {
	const YEAR = 2018;
	const MONTH_INDEX = 5; // month July

	beforeEach(() => {
		mockIsTodayWith(new Date(YEAR, MONTH_INDEX, 20));
	});

	it('should render a DatePicker', () => {
		// given
		const calendar = { year: YEAR, monthIndex: MONTH_INDEX };
		const isDisabledChecker = getDisabledChecker([
			new Date(YEAR, MONTH_INDEX, 6),
			new Date(YEAR, MONTH_INDEX, 15),
		]);
		const selectedDate = new Date(YEAR, MONTH_INDEX, 12);

		// when
		const { container } = render(
			<DatePicker
				calendar={calendar}
				isDisabledChecker={isDisabledChecker}
				onSelect={jest.fn()}
				goToPreviousMonth={jest.fn()}
				goToNextMonth={jest.fn()}
				selectedDate={selectedDate}
			/>,
		);

		// then
		expect(container.firstChild).toMatchSnapshot();
	});

	it('should highlight today', () => {
		// given
		const calendar = { year: YEAR, monthIndex: MONTH_INDEX };

		// when
		render(
			<DatePicker
				calendar={calendar}
				onSelect={jest.fn()}
				goToPreviousMonth={jest.fn()}
				goToNextMonth={jest.fn()}
			/>,
		);

		// then
		expect(screen.getByText('20')).toHaveClass('theme-today');
	});

	it('should highlight selected date', () => {
		// given
		const calendar = { year: YEAR, monthIndex: MONTH_INDEX };
		const selectedDate = new Date(YEAR, MONTH_INDEX, 12);

		// when
		render(
			<DatePicker
				calendar={calendar}
				onSelect={jest.fn()}
				goToPreviousMonth={jest.fn()}
				goToNextMonth={jest.fn()}
				selectedDate={selectedDate}
			/>,
		);

		// then
		expect(screen.getByText('12')).toHaveClass('theme-selected');
	});

	it('should disable date be a disabled button', () => {
		// given
		const calendar = { year: YEAR, monthIndex: MONTH_INDEX };
		const isDisabledChecker = getDisabledChecker([new Date(YEAR, MONTH_INDEX, 6)]);

		// when
		render(
			<DatePicker
				calendar={calendar}
				isDisabledChecker={isDisabledChecker}
				onSelect={jest.fn()}
				goToPreviousMonth={jest.fn()}
				goToNextMonth={jest.fn()}
			/>,
		);

		// then
		expect(screen.getAllByText('6')[0]).toBeDisabled();
	});

	it('should select date', async () => {
		const user = userEvent.setup();

		// given
		const calendar = { year: YEAR, monthIndex: MONTH_INDEX };
		const onSelect = jest.fn();
		render(
			<DatePicker
				calendar={calendar}
				onSelect={onSelect}
				goToPreviousMonth={jest.fn()}
				goToNextMonth={jest.fn()}
			/>,
		);
		expect(onSelect).not.toHaveBeenCalled();

		// when
		await user.click(screen.getAllByText('1')[0]);

		// then
		expect(onSelect).toHaveBeenCalledWith(expect.anything(), new Date(YEAR, MONTH_INDEX, 1));
	});

	it('should manage tabIndex', () => {
		const calendar = { year: YEAR, monthIndex: MONTH_INDEX };
		const { rerender } = render(
			<DatePicker
				calendar={calendar}
				onSelect={jest.fn()}
				goToPreviousMonth={jest.fn()}
				goToNextMonth={jest.fn()}
			/>,
		);
		expect(screen.getAllByText('1')[0]).toHaveAttribute('tabIndex', '-1');

		// when
		rerender(
			<DatePicker
				calendar={calendar}
				onSelect={jest.fn()}
				goToPreviousMonth={jest.fn()}
				goToNextMonth={jest.fn()}
				allowFocus
			/>,
		);

		// then
		expect(screen.getAllByText('1')[0]).toHaveAttribute('tabIndex', '0');
	});

	it('should have 6 weeks', () => {
		const calendar = { year: YEAR, monthIndex: MONTH_INDEX };
		render(
			<DatePicker
				calendar={calendar}
				onSelect={jest.fn()}
				goToPreviousMonth={jest.fn()}
				goToNextMonth={jest.fn()}
			/>,
		);
		expect(screen.getAllByRole('button')).toHaveLength(6 * 7);
	});

	it('should go to next month if select a date of next month', async () => {
		const year = 2019;
		const monthIndex = 11;
		const calendar = { year, monthIndex };

		const props = {
			calendar,
			onSelect: jest.fn(),
			goToPreviousMonth: jest.fn(),
			goToNextMonth: jest.fn(),
		};
		render(<DatePicker {...props} />);
		await userEvent.click(screen.getByLabelText('Saturday 04 January 2020'));
		expect(props.onSelect).toHaveBeenCalledWith(expect.anything(), new Date(year + 1, 0, 4));
		expect(props.goToNextMonth).toHaveBeenCalled();
		expect(props.goToPreviousMonth).not.toHaveBeenCalled();
	});
});
