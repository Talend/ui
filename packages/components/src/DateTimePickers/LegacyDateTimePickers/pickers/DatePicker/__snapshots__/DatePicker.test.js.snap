// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker should render a DatePicker 1`] = `
<CalendarGesture(DatePicker)
  calendar={
    Object {
      "monthIndex": 5,
      "year": 2018,
    }
  }
  goToNextMonth={[MockFunction]}
  goToPreviousMonth={[MockFunction]}
  isDisabledChecker={[Function]}
  onSelect={[MockFunction]}
  selectedDate={2018-06-11T22:00:00.000Z}
>
  <div>
    <DatePicker
      calendar={
        Object {
          "monthIndex": 5,
          "year": 2018,
        }
      }
      goToNextMonth={[MockFunction]}
      goToPreviousMonth={[MockFunction]}
      isDisabledChecker={[Function]}
      onKeyDown={[Function]}
      onSelect={[MockFunction]}
      selectedDate={2018-06-11T22:00:00.000Z}
      t={[Function]}
    >
      <table
        className="theme-container"
      >
        <caption
          className="sr-only"
        >
          June 2018
        </caption>
        <thead>
          <tr
            className="theme-calendar-header"
          >
            <th
              key="0"
              scope="col"
            >
              <abbr
                key="0"
                title="Monday"
              >
                M
              </abbr>
            </th>
            <th
              key="1"
              scope="col"
            >
              <abbr
                key="1"
                title="Tuesday"
              >
                T
              </abbr>
            </th>
            <th
              key="2"
              scope="col"
            >
              <abbr
                key="2"
                title="Wednesday"
              >
                W
              </abbr>
            </th>
            <th
              key="3"
              scope="col"
            >
              <abbr
                key="3"
                title="Thursday"
              >
                T
              </abbr>
            </th>
            <th
              key="4"
              scope="col"
            >
              <abbr
                key="4"
                title="Friday"
              >
                F
              </abbr>
            </th>
            <th
              key="5"
              scope="col"
            >
              <abbr
                key="5"
                title="Saturday"
              >
                S
              </abbr>
            </th>
            <th
              key="6"
              scope="col"
            >
              <abbr
                key="6"
                title="Sunday"
              >
                S
              </abbr>
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            className="theme-calendar-row tc-date-picker-calendar-row"
            key="0"
          >
            <td
              className="theme-calendar-col"
              key="0"
            >
              <button
                aria-label="Monday 28 May 2018"
                className="theme-calendar-day theme-not-current-month tc-date-picker-day btn-tertiary btn-default"
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                28
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="1"
            >
              <button
                aria-label="Tuesday 29 May 2018"
                className="theme-calendar-day theme-not-current-month tc-date-picker-day btn-tertiary btn-default"
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                29
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="2"
            >
              <button
                aria-label="Wednesday 30 May 2018"
                className="theme-calendar-day theme-not-current-month tc-date-picker-day btn-tertiary btn-default"
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                30
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="3"
            >
              <button
                aria-label="Thursday 31 May 2018"
                className="theme-calendar-day theme-not-current-month tc-date-picker-day btn-tertiary btn-default"
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                31
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="4"
            >
              <button
                aria-label="Friday 01 June 2018"
                className="theme-calendar-day tc-date-picker-day btn-tertiary btn-default"
                data-value={1}
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                1
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="5"
            >
              <button
                aria-label="Saturday 02 June 2018"
                className="theme-calendar-day tc-date-picker-day btn-tertiary btn-default"
                data-value={2}
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                2
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="6"
            >
              <button
                aria-label="Sunday 03 June 2018"
                className="theme-calendar-day tc-date-picker-day btn-tertiary btn-default"
                data-value={3}
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                3
              </button>
            </td>
          </tr>
          <tr
            className="theme-calendar-row tc-date-picker-calendar-row"
            key="1"
          >
            <td
              className="theme-calendar-col"
              key="0"
            >
              <button
                aria-label="Monday 04 June 2018"
                className="theme-calendar-day tc-date-picker-day btn-tertiary btn-default"
                data-value={4}
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                4
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="1"
            >
              <button
                aria-label="Tuesday 05 June 2018"
                className="theme-calendar-day tc-date-picker-day btn-tertiary btn-default"
                data-value={5}
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                5
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="2"
            >
              <button
                aria-label="Wednesday 06 June 2018"
                className="theme-calendar-day tc-date-picker-day btn-tertiary btn-default"
                data-value={6}
                disabled={true}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                6
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="3"
            >
              <button
                aria-label="Thursday 07 June 2018"
                className="theme-calendar-day tc-date-picker-day btn-tertiary btn-default"
                data-value={7}
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                7
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="4"
            >
              <button
                aria-label="Friday 08 June 2018"
                className="theme-calendar-day tc-date-picker-day btn-tertiary btn-default"
                data-value={8}
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                8
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="5"
            >
              <button
                aria-label="Saturday 09 June 2018"
                className="theme-calendar-day tc-date-picker-day btn-tertiary btn-default"
                data-value={9}
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                9
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="6"
            >
              <button
                aria-label="Sunday 10 June 2018"
                className="theme-calendar-day tc-date-picker-day btn-tertiary btn-default"
                data-value={10}
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                10
              </button>
            </td>
          </tr>
          <tr
            className="theme-calendar-row tc-date-picker-calendar-row"
            key="2"
          >
            <td
              className="theme-calendar-col"
              key="0"
            >
              <button
                aria-label="Monday 11 June 2018"
                className="theme-calendar-day tc-date-picker-day btn-tertiary btn-default"
                data-value={11}
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                11
              </button>
            </td>
            <td
              aria-current="date"
              className="theme-calendar-col"
              key="1"
            >
              <button
                aria-label="Tuesday 12 June 2018, selected"
                className="theme-calendar-day theme-selected tc-date-picker-day btn-tertiary btn-default"
                data-value={12}
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                12
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="2"
            >
              <button
                aria-label="Wednesday 13 June 2018"
                className="theme-calendar-day tc-date-picker-day btn-tertiary btn-default"
                data-value={13}
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                13
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="3"
            >
              <button
                aria-label="Thursday 14 June 2018"
                className="theme-calendar-day tc-date-picker-day btn-tertiary btn-default"
                data-value={14}
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                14
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="4"
            >
              <button
                aria-label="Friday 15 June 2018"
                className="theme-calendar-day tc-date-picker-day btn-tertiary btn-default"
                data-value={15}
                disabled={true}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                15
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="5"
            >
              <button
                aria-label="Saturday 16 June 2018"
                className="theme-calendar-day tc-date-picker-day btn-tertiary btn-default"
                data-value={16}
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                16
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="6"
            >
              <button
                aria-label="Sunday 17 June 2018"
                className="theme-calendar-day tc-date-picker-day btn-tertiary btn-default"
                data-value={17}
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                17
              </button>
            </td>
          </tr>
          <tr
            className="theme-calendar-row tc-date-picker-calendar-row"
            key="3"
          >
            <td
              className="theme-calendar-col"
              key="0"
            >
              <button
                aria-label="Monday 18 June 2018"
                className="theme-calendar-day tc-date-picker-day btn-tertiary btn-default"
                data-value={18}
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                18
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="1"
            >
              <button
                aria-label="Tuesday 19 June 2018"
                className="theme-calendar-day tc-date-picker-day btn-tertiary btn-default"
                data-value={19}
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                19
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="2"
            >
              <button
                aria-label="Today, Wednesday 20 June 2018"
                className="theme-calendar-day theme-today tc-date-picker-day btn-tertiary btn-default"
                data-value={20}
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                20
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="3"
            >
              <button
                aria-label="Thursday 21 June 2018"
                className="theme-calendar-day tc-date-picker-day btn-tertiary btn-default"
                data-value={21}
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                21
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="4"
            >
              <button
                aria-label="Friday 22 June 2018"
                className="theme-calendar-day tc-date-picker-day btn-tertiary btn-default"
                data-value={22}
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                22
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="5"
            >
              <button
                aria-label="Saturday 23 June 2018"
                className="theme-calendar-day tc-date-picker-day btn-tertiary btn-default"
                data-value={23}
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                23
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="6"
            >
              <button
                aria-label="Sunday 24 June 2018"
                className="theme-calendar-day tc-date-picker-day btn-tertiary btn-default"
                data-value={24}
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                24
              </button>
            </td>
          </tr>
          <tr
            className="theme-calendar-row tc-date-picker-calendar-row"
            key="4"
          >
            <td
              className="theme-calendar-col"
              key="0"
            >
              <button
                aria-label="Monday 25 June 2018"
                className="theme-calendar-day tc-date-picker-day btn-tertiary btn-default"
                data-value={25}
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                25
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="1"
            >
              <button
                aria-label="Tuesday 26 June 2018"
                className="theme-calendar-day tc-date-picker-day btn-tertiary btn-default"
                data-value={26}
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                26
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="2"
            >
              <button
                aria-label="Wednesday 27 June 2018"
                className="theme-calendar-day tc-date-picker-day btn-tertiary btn-default"
                data-value={27}
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                27
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="3"
            >
              <button
                aria-label="Thursday 28 June 2018"
                className="theme-calendar-day tc-date-picker-day btn-tertiary btn-default"
                data-value={28}
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                28
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="4"
            >
              <button
                aria-label="Friday 29 June 2018"
                className="theme-calendar-day tc-date-picker-day btn-tertiary btn-default"
                data-value={29}
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                29
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="5"
            >
              <button
                aria-label="Saturday 30 June 2018"
                className="theme-calendar-day tc-date-picker-day btn-tertiary btn-default"
                data-value={30}
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                30
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="6"
            >
              <button
                aria-label="Sunday 01 July 2018"
                className="theme-calendar-day theme-not-current-month tc-date-picker-day btn-tertiary btn-default"
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                1
              </button>
            </td>
          </tr>
          <tr
            className="theme-calendar-row tc-date-picker-calendar-row"
            key="5"
          >
            <td
              className="theme-calendar-col"
              key="0"
            >
              <button
                aria-label="Monday 02 July 2018"
                className="theme-calendar-day theme-not-current-month tc-date-picker-day btn-tertiary btn-default"
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                2
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="1"
            >
              <button
                aria-label="Tuesday 03 July 2018"
                className="theme-calendar-day theme-not-current-month tc-date-picker-day btn-tertiary btn-default"
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                3
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="2"
            >
              <button
                aria-label="Wednesday 04 July 2018"
                className="theme-calendar-day theme-not-current-month tc-date-picker-day btn-tertiary btn-default"
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                4
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="3"
            >
              <button
                aria-label="Thursday 05 July 2018"
                className="theme-calendar-day theme-not-current-month tc-date-picker-day btn-tertiary btn-default"
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                5
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="4"
            >
              <button
                aria-label="Friday 06 July 2018"
                className="theme-calendar-day theme-not-current-month tc-date-picker-day btn-tertiary btn-default"
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                6
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="5"
            >
              <button
                aria-label="Saturday 07 July 2018"
                className="theme-calendar-day theme-not-current-month tc-date-picker-day btn-tertiary btn-default"
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                7
              </button>
            </td>
            <td
              className="theme-calendar-col"
              key="6"
            >
              <button
                aria-label="Sunday 08 July 2018"
                className="theme-calendar-day theme-not-current-month tc-date-picker-day btn-tertiary btn-default"
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                8
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </DatePicker>
  </div>
</CalendarGesture(DatePicker)>
`;
